File: LeetCode/problems/2485-find-the-pivot-integer/2485-find-the-pivot-integer.py
class Solution:
    def pivotInteger(self,n):
        if n == 1:
            return 1

        sum_total = (n + 1) * (n // 2)
        temp = 0

        for i in range(n):
            sum_total -= i
            if sum_total == temp:
                return i
            temp += i

        return -1

File: LeetCode/problems/2485-find-the-pivot-integer/2485-find-the-pivot-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var pivotInteger = function(n) {
    const AS = (x)=>(x/2)*(x+1);
    if(n==1){return 1;}
    for(let I = 1; I<n; I++){
        if( AS(I) == AS(n)-AS(I-1)){
            return I;
        }
    }
    return -1;
};
